src/Dapr.PluggableComponents.AspNetCore/WebApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,107 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Collections.Concurrent;
+using System.Globalization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Dapr.PluggableComponents;
+
+/// <summary>
+/// Represents extension methods for registering Dapr Pluggable Component related services with <see cref="WebApplicationBuilder"/> instances.
+/// </summary>
+public static class WebApplicationBuilderExtensions
+{
+    private static readonly ConcurrentDictionary<string, bool> SocketPaths = new ConcurrentDictionary<string, bool>();
+
+    /// <summary>
+    /// Registers services needed to host Dapr Pluggable Components via an ASP.NET application.
+    /// </summary>
+    /// <param name="builder">A <see cref="WebApplicationBuilder"/> instance.</param>
+    /// <returns>The current <see cref="WebApplicationBuilder"/> instance.</returns>
+    public static WebApplicationBuilder AddDaprPluggableComponentsSupportServices(this WebApplicationBuilder builder)
+    {
+        builder.Services.AddGrpc();
+
+        // Dapr component discovery relies on the gRPC reflection service.
+        builder.Services.AddGrpcReflection();
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Registers a Dapr "service" (i.e. a Unix domain socket via which Dapr Pluggable Components can be invoked).
+    /// </summary>
+    /// <param name="builder">A <see cref="WebApplicationBuilder"/> instance.</param>
+    /// <param name="options">Options related to the creation of the socket file.</param>
+    /// <returns>The full path to the socketfile representing the service.</returns>
+    /// <exception cref="ArgumentException">Thrown if the socket file path cannot be built or has already been registered. </exception>
+    public static string AddDaprService(this WebApplicationBuilder builder, DaprPluggableComponentsServiceOptions options)
```

Comment:
Should this just add the support services itself?

Reply:
This method is intended to be called multiple times, which would then cause those services to be added multiple times.  While I *fully expect* those service registration methods to reasonably handle such scenarios, as I don't own the code I can't *guarantee* that.  Hence, I'd say it's safer to keep them separate.

As these extension methods are intended for use by us, primarily, we can ensure the support services are added (and only once).  In cases where end users call this method, in setting up a customized Dapr pluggable component compatible environment beyond our happy path, I'd say the bar is a little higher in terms of understanding how these methods are expected to be used.